backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/cgen/CBridgeGen.kt

```diff
@@ -1410,6 +1446,23 @@ private object VoidReturning : ValueReturning {
     }
 }
 
+private object IgnoredUnitArgumentPassing : ArgumentPassing {
+    override fun KotlinToCCallBuilder.passValue(expression: IrExpression): CExpression? {
+        // Note: it is not correct to just drop the expression (due to possible side effects),
+        // so (in lack of other options) evaluate the expression and pass ignored value to the bridge:
+        val bridgeArgument = irBuilder.irBlock {
+            +expression
+            +irInt(0)
+        }
+        passThroughBridge(bridgeArgument, irBuilder.context.irBuiltIns.intType, CTypes.int).name
```

Comment:
Do we really need to get `name` here?
Also why do we pass an ignored `int` into the bridge? Why can't we pass `expression` directly?

Reply:
Discussed offline.